packages/tokamak/contracts-bedrock/src/L1/L1StandardBridge.sol

```diff
@@ -236,22 +337,80 @@ contract L1StandardBridge is StandardBridge, ISemver {
         _initiateBridgeERC20(_l1Token, _l2Token, _from, _to, _amount, _minGasLimit, _extraData);
     }
 
-    /// @inheritdoc StandardBridge
-    /// @notice Emits the legacy ETHDepositInitiated event followed by the ETHBridgeInitiated event.
-    ///         This is necessary for backwards compatibility with the legacy bridge.
-    function _emitETHBridgeInitiated(
+    /// @notice Sends ERC20 tokens to a receiver's address on the other chain.
+    /// @param _localToken  Address of the ERC20 on this chain.
+    /// @param _remoteToken Address of the corresponding token on the remote chain.
+    /// @param _to          Address of the receiver.
+    /// @param _amount      Amount of local tokens to deposit.
+    /// @param _minGasLimit Minimum amount of gas that the bridge can be relayed with.
+    /// @param _extraData   Extra data to be sent with the transaction. Note that the recipient will
+    ///                     not be triggered with this data, but it will be emitted and can be used
+    ///                     to identify the transaction.
+    function _initiateBridgeERC20(
+        address _localToken,
+        address _remoteToken,
         address _from,
         address _to,
         uint256 _amount,
+        uint32 _minGasLimit,
         bytes memory _extraData
     )
         internal
         override
     {
-        emit ETHDepositInitiated(_from, _to, _amount, _extraData);
-        super._emitETHBridgeInitiated(_from, _to, _amount, _extraData);
+        if (l1TONAddress == _localToken) {
+            _initiateBridgeTON(_from, _to, _amount, _minGasLimit, _extraData);
+        } else {
+            super._initiateBridgeERC20(_localToken, _remoteToken, _from, _to, _amount, _minGasLimit, _extraData);
+        }
+    }
+
+    /// @notice Sends TON tokens to a receiver's address on the other chain.
+    /// @param _to          Address of the receiver.
+    /// @param _amount      Amount of local tokens to deposit.
+    /// @param _minGasLimit Minimum amount of gas that the bridge can be relayed with.
+    /// @param _extraData   Extra data to be sent with the transaction. Note that the recipient will
+    ///                     not be triggered with this data, but it will be emitted and can be used
+    ///                     to identify the transaction.
+    function _initiateBridgeTON(
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes memory _extraData
+    )
+        internal
+    {
+        IERC20(l1TONAddress).safeTransferFrom(_from, address(this), _amount);
```

Comment:
Here's the code I'm curious about.

Reply:
It is called when users deposit TON in L1StandardBidge. Basically, this line will take user's token to L1StandardBridge. After that, the token is sent to L1CrossDomainMessenger and then forward to OptimismPortal

Reply:
thank you very much brave! i missed function(L1CrossDomainMessenger:_sendMessge line:67~69)

this function(L1CrossDomainMessenger:_sendMessge) transfer ton L1Bridge to L1XDomainMessenger